src/interpreter/executors/main.rs

```diff
@@ -84,13 +84,21 @@ where
                 }
             }
 
-            Transaction::Script { .. } => {
+            Transaction::Script { inputs, .. } => {
+                if inputs.iter().any(|input| {
+                    if let Input::Contract { contract_id, .. } = input {
+                        !self.check_contract_exists(contract_id).unwrap_or(false)
+                    } else {
+                        false
+                    }
+                }) {
+                    return Err(InterpreterError::Panic(PanicReason::ContractNotFound));
+                }
+
                 let offset = (VM_TX_MEMORY + Transaction::script_offset()) as Word;
 
                 self.registers[REG_PC] = offset;
                 self.registers[REG_IS] = offset;
-                self.registers[REG_GGAS] = self.tx.gas_limit();
-                self.registers[REG_CGAS] = self.tx.gas_limit();
```

Comment:
Why were these removed?

Reply:
Because it's duplicating logic done in `initialization.rs`